src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -83,43 +82,16 @@ public void remove(T toRemove) {
     }
 
     /**
-     * Retrieves one of the elderly with the given nric from the list.
-     * The elderly with such a nric must exist in the list.
+     * Retrieves one of the persons with the given nric from the list.
+     * The person with such a nric must exist in the list.
      *
-     * @param nric Nric of the elderly.
-     * @return Elderly with that name.
+     * @param nric Nric of the person.
+     * @return Person with that name.
      */
-    public Elderly getElderly(Nric nric) {
+    public T get(Nric nric) {
         requireNonNull(nric);
-        for (Person person : this) {
-            if (person instanceof Elderly) {
-                Elderly elderly = (Elderly) person;
-                if (elderly.getNric().equals(nric)) {
-                    return elderly;
-                }
-            }
-        }
-        throw new ElderlyNotFoundException();
-    }
-
-    /**
-     * Retrieves one of the volunteer with the given nric from the list.
-     * The volunteer with such a nric must exist in the list.
-     *
-     * @param nric Nric of the volunteer.
-     * @return volunteer with that name.
-     */
-    public Volunteer getVolunteer(Nric nric) {
-        requireNonNull(nric);
-        for (Person person : this) {
-            if (person instanceof Volunteer) {
-                Volunteer volunteer = (Volunteer) person;
-                if (volunteer.getNric().equals(nric)) {
-                    return volunteer;
-                }
-            }
-        }
-        throw new VolunteerNotFoundException();
+        FilteredList<T> filteredList = internalList.filtered((T person) -> person.getNric().equals(nric));
+        return filteredList.size() == 0 ? null : filteredList.get(0);
     }
```

Comment:
So now no separate getElderly and getVolunteer? Will this affect some other functions that call these methods, or is it better we still keep getElderly and getVolunteer, that call this get() method internally?

Reply:
I've checked through the code to make sure that this change is safe. From an implementation point of view, I think it makes more sense to have this `get` method as opposed to the `getElderly` and `getVolunteer` methods.